### ISSUE TITLE
[haproxy] Unable to build on arm

### ISSUE THREAD

**Issue by uberhacker** [skycocker/chromebrew#3339]
```
/usr/local/bin/ld: src/fd.o: in function `fd_rm_from_fd_list':
/usr/local/tmp/crew/haproxy-1.9.8.tar.gz.dir/haproxy-1.9.8/src/fd.c:267: undefined reference to `_HA_ATOMIC_DWCAS'
collect2: error: ld returned 1 exit status
make: *** [Makefile:994: haproxy] Error 1
haproxy failed to build: `make V=0 -j4 TARGET=linux26 USE_OPENSSL=1 USE_PCRE=1 USE_ZLIB=1` exited with 2
```

**Comment by JL2210** [comment]
Can you give more of the log above this? Try to look for red, purple, or blue text, as those usually indicate compiler warnings.

**Comment by uberhacker** [comment]
The error is `haproxy-1.9.8/src/fd.c:267: undefined reference to `_HA_ATOMIC_DWCAS'`.

**Comment by JL2210** [comment]
I understand. What are the **compiler warnings** emitted? Not linker errors.

**Comment by JL2210** [comment]
Also, what is the compiler version (`gcc --version`)?

**Comment by cstrouse** [comment]
```
No precompiled binary available for your platform, downloading source...
############################################################################################################################################################### 100.0%
Archive downloaded
Unpacking archive using 'tar', this may take a while...
gcc (GCC) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Building from source, this may take a while...
Rename all *.la files to *.la_tmp
  CC      src/ev_poll.o
  CC      src/ev_epoll.o
  CC      src/ssl_sock.o
  CC      src/proto_http.o
  CC      src/cfgparse-listen.o
  CC      src/proto_htx.o
  CC      src/stream.o
  CC      src/mux_h2.o
  CC      src/stats.o
  CC      src/flt_spoe.o
  CC      src/server.o
  CC      src/checks.o
  CC      src/haproxy.o
  CC      src/cfgparse.o
  CC      src/flt_http_comp.o
  CC      src/http_fetch.o
  CC      src/dns.o
  CC      src/stick_table.o
  CC      src/mux_h1.o
  CC      src/peers.o
  CC      src/standard.o
  CC      src/proxy.o
  CC      src/cli.o
  CC      src/log.o
  CC      src/backend.o
  CC      src/pattern.o
  CC      src/sample.o
  CC      src/stream_interface.o
  CC      src/proto_tcp.o
  CC      src/listener.o
  CC      src/h1.o
  CC      src/cfgparse-global.o
  CC      src/cache.o
  CC      src/http_rules.o
  CC      src/http_act.o
  CC      src/tcp_rules.o
  CC      src/filters.o
  CC      src/connection.o
  CC      src/session.o
  CC      src/acl.o
  CC      src/vars.o
  CC      src/raw_sock.o
  CC      src/map.o
  CC      src/proto_uxst.o
  CC      src/payload.o
  CC      src/fd.o
  CC      src/queue.o
In file included from include/common/config.h:25,
                 from src/fd.c:154:
src/fd.c: In function ‘fd_rm_from_fd_list’:
src/fd.c:267:16: warning: implicit declaration of function ‘_HA_ATOMIC_DWCAS’; did you mean ‘HA_ATOMIC_DWCAS’? [-Wimplicit-function-declaration]
      unlikely(!_HA_ATOMIC_DWCAS(((void **)(void *)&_GET_NEXT(fd, off)), ((void **)(void *)&cur_list), (*(void **)(void *)&next_list))))
                ^~~~~~~~~~~~~~~~
include/common/compiler.h:125:40: note: in definition of macro ‘unlikely’
 #define unlikely(x) (__builtin_expect((x) != 0, 0))
                                        ^
  CC      src/flt_trace.o
  CC      src/task.o
  CC      src/lb_chash.o
  CC      src/frontend.o
  CC      src/applet.o
  CC      src/mux_pt.o
  CC      src/signal.o
  CC      src/ev_select.o
  CC      src/proto_sockpair.o
  CC      src/compression.o
  CC      src/http_conv.o
  CC      src/memory.o
  CC      src/lb_fwrr.o
  CC      src/channel.o
  CC      src/htx.o
  CC      src/uri_auth.o
  CC      src/regex.o
  CC      src/chunk.o
  CC      src/pipe.o
  CC      src/lb_fas.o
  CC      src/lb_map.o
  CC      src/lb_fwlc.o
  CC      src/auth.o
  CC      src/time.o
  CC      src/hathreads.o
  CC      src/http_htx.o
  CC      src/buffer.o
  CC      src/hpack-tbl.o
  CC      src/shctx.o
  CC      src/sha1.o
  CC      src/http.o
  CC      src/hpack-dec.o
  CC      src/action.o
  CC      src/proto_udp.o
  CC      src/http_acl.o
  CC      src/xxhash.o
  CC      src/hpack-enc.o
  CC      src/h2.o
  CC      src/freq_ctr.o
  CC      src/lru.o
  CC      src/protocol.o
  CC      src/arg.o
  CC      src/hpack-huff.o
  CC      src/hdr_idx.o
  CC      src/base64.o
  CC      src/hash.o
  CC      src/mailers.o
  CC      src/activity.o
  CC      src/http_msg.o
  CC      ebtree/ebtree.o
  CC      ebtree/eb32sctree.o
  CC      ebtree/eb32tree.o
  CC      ebtree/eb64tree.o
  CC      ebtree/ebmbtree.o
  CC      ebtree/ebsttree.o
  CC      ebtree/ebimtree.o
  CC      ebtree/ebistree.o
  LD      haproxy
/usr/local/bin/ld: src/fd.o: in function `fd_rm_from_fd_list':
/usr/local/tmp/crew/haproxy-1.9.8.tar.gz.dir/haproxy-1.9.8/src/fd.c:267: undefined reference to `_HA_ATOMIC_DWCAS'
collect2: error: ld returned 1 exit status
make: *** [Makefile:994: haproxy] Error 1
haproxy failed to install: `make V=0 -j6 TARGET=linux26 USE_OPENSSL=1 USE_PCRE=1 USE_ZLIB=1` exited with 2
```
```
chronos@localhost / $ gcc --version
gcc (GCC) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by JL2210** [comment]
@cstrouse Thank you. I'll look into it.

**Comment by JL2210** [comment]
See haproxy/haproxy#105. This probably requires an update.

**Comment by uberhacker** [comment]
Fixed in #3384
